typescript/core/src/v0/ChunkCursor.ts

```diff
@@ -64,65 +64,18 @@ export class ChunkCursor {
       throw new Error("Cannot compare a reversed ChunkCursor to a non-reversed ChunkCursor");
     }
 
-    if (reverse) {
-      // If chunks don't overlap at all, sort later chunk first
-      if (this.chunkIndex.messageEndTime < other.chunkIndex.messageStartTime) {
-        return 1;
-      }
-      if (this.chunkIndex.messageStartTime > other.chunkIndex.messageEndTime) {
-        return -1;
-      }
-    } else {
-      // If chunks don't overlap at all, sort earlier chunk first
-      if (this.chunkIndex.messageEndTime < other.chunkIndex.messageStartTime) {
-        return -1;
-      }
-      if (this.chunkIndex.messageStartTime > other.chunkIndex.messageEndTime) {
-        return 1;
+    const diff = Number(this.getSortTime() - other.getSortTime());
+    if (diff === 0) {
+      if (reverse) {
+        // Break ties by chunk offset in the file
+        return Number(other.chunkIndex.chunkStartOffset - this.chunkIndex.chunkStartOffset);
+      } else {
+        // Break ties by chunk offset in the file
+        return Number(this.chunkIndex.chunkStartOffset - other.chunkIndex.chunkStartOffset);
```

Comment:
I might be jumping in with too little context here, but I think sort stability is going to depend on fixing,
* message log time
* offset of message within chunk
* offset of chunk within file

AFAICT the tie-breaking is only considering the first and third of those, but not the second. So if you had two messages with the same timestamp in the same chunk, what guarantees stability?

Reply:
@wkalt This is comparing two different chunks - if you have two messages with the same timestamp in the same chunk those are in the same chunk and don't impact sorting one chunk against another.